bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/obsComp/PwfsGuideProbe.java

```diff
@@ -284,41 +283,48 @@ public GuideStarValidation validate(Coordinates coords, ObsContext ctx) {
         return GuideProbeUtil.instance.validate(coords, this, ctx);
     }
 
-    public Option<BoundaryPosition> checkBoundaries(SPTarget guideStar, ObsContext ctx){
+    public Option<PwfsProbeRangeArea> checkBoundaries(SPTarget guideStar, ObsContext ctx){
         return guideStar
             .getSkycalcCoordinates(ctx.getSchedulingBlockStart())
             .flatMap(cs -> checkBoundaries(cs, ctx));
     }
 
     /**
-     * Returns BoundaryPosition.outside, if the coordinates are outside the outer limit,
-     * inside, if inside the inner limit and innerBoundary if between the inner and outer limits
-     * (which is where the PWFS should normally be).
+     * Returns:
+     *
+     * - BoundaryCheck.outOfRange if outside the limit.
+     *
+     * - BoundaryCheck.vignetting if it obscures the science area.
+     *
+     * - BoundaryCheck.inRange if inside the limit.
      *
      * @param coords the coordinates
      * @param ctx the context
      */
-    public Option<BoundaryPosition> checkBoundaries(final Coordinates coords, final ObsContext ctx) {
+    public Option<PwfsProbeRangeArea> checkBoundaries(final Coordinates coords, final ObsContext ctx) {
         return ctx.getBaseCoordinates().map(baseCoordinates -> {
             final Angle positionAngle = ctx.getPositionAngleJava();
             final Set<Offset> sciencePositions = ctx.getSciencePositions();
 
             // check positions against corrected outer patrol field bounds
             return getCorrectedPatrolField(ctx).map(pf -> {
                 final BoundaryPosition bp = pf.checkBoundaries(coords, baseCoordinates, positionAngle, sciencePositions);
-                if (bp != BoundaryPosition.inside) {
-                    return BoundaryPosition.outside;
-                }
-
-                // Check if any of the guide stars are inside the inner bounds (opposite logic needed, union instead of intersection)
-                final double minLimit = getVignettingClearance(ctx).toArcsecs().getMagnitude();
-                final Ellipse2D e = new Ellipse2D.Double(-minLimit, -minLimit, minLimit * 2.0, minLimit * 2.0);
-                final PatrolField p = getCorrectedPatrolField(new PatrolField(e, e, e), ctx);
-                if (p.anyInside(coords, baseCoordinates, positionAngle, sciencePositions)) {
-                    return BoundaryPosition.inside;
+                switch (bp) {
+                    case outside:
+                    case outerBoundary:
+                        return PwfsProbeRangeArea.outOfRange;
+                    case inside:
+                    case innerBoundary:
```

Comment:
It can't be both `innerBoundary` and in the vignetting area so you could short-circuit this case a bit and leave the vignetting check for `inside`.  On the other hand, it is rare that a guide star falls in that tiny `innerBoundary` area anyway.